### ISSUE TITLE
Supervised Contrastive Learning for Pre-trained Language Model Fine-tuning

### ISSUE THREAD

**Issue by morioka** [morioka/reading#35]
https://twitter.com/arxiv_org/status/1324123219612246018

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Supervised Contrastive Learning for Pre-trained Language Model Fine-tuning. <a href="https://t.co/WIYOoid5Rb">https://t.co/WIYOoid5Rb</a> <a href="https://t.co/kUCcuYvZMP">pic.twitter.com/kUCcuYvZMP</a></p>&mdash; arxiv (@arxiv_org) <a href="https://twitter.com/arxiv_org/status/1324123219612246018?ref_src=twsrc%5Etfw">November 4, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

https://arxiv.org/abs/2011.01403

Supervised Contrastive Learning for Pre-trained Language Model Fine-tuning

Beliz Gunel, Jingfei Du, Alexis Conneau, Ves Stoyanov

Stanford,  FAIR

State-of-the-art natural language understanding classification models follow two-stages: pre-training a large language model on an auxiliary task, and then fine-tuning the model on a task-specific labeled dataset using cross-entropy loss. Cross-entropy loss has several shortcomings that can lead to sub-optimal generalization and instability. Driven by the intuition that good generalization requires capturing the similarity between examples in one class and contrasting them with examples in other classes, we propose a supervised contrastive learning (SCL) objective for the fine-tuning stage. Combined with cross-entropy, the SCL loss we propose obtains improvements over a strong RoBERTa-Large baseline on multiple datasets of the GLUE benchmark in both the high-data and low-data regimes, and it does not require any specialized architecture, data augmentation of any kind, memory banks, or additional unsupervised data. We also demonstrate that the new objective leads to models that are more robust to different levels of noise in the training data, and can generalize better to related tasks with limited labeled task data.

![image](https://user-images.githubusercontent.com/1615546/98460751-6ece2680-21ea-11eb-9d89-ac96cd4c3c0e.png)
![image](https://user-images.githubusercontent.com/1615546/98460763-85747d80-21ea-11eb-99f2-07bedf55117b.png)

**Comment by DeepTecher** [comment]
have u found the implement?

**Comment by morioka** [comment]
not yet.
